docs/manual/about/release-process.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Release process
+description: Vector's release process. Covering cadence and compatibility guarantees.
+---
+
+Here you can find information about how to track Vector's releases, how often
+we release, compatibility guarantees, and more.
+
+## How to track releases
+
+There are a few different ways to track when there is a new release:
+
+* Following [@vectordotdev](https://twitter.com/vectordotdev) on Twitter.
```

Comment:
Nit: in this list I think "Follow" etc works a bit better than "Following" etc. The Gerund Police are gonna let you off with a very light warning 👮🏼 THIS TIME.